nio/block/mixins/persistence/persistence.py

```diff
@@ -47,24 +47,40 @@ def _load(self):
         """
         self.logger.debug("Loading from persistence")
         # load whole item from persistence
-        item = self._persistence.load(self.id(), default={})
-        for persisted_var in self.persisted_values():
-            if persisted_var in item:
-                self.logger.debug("Loaded value {} for attribute {}".format(
-                    item[persisted_var], persisted_var))
-                # Set the loaded value to the attribute on this class
-                setattr(self, persisted_var, item[persisted_var])
+        data = self._persistence.load(self.id(), default={})
+        if not data:
+            return
+
+        try:
+            self.deserialize(data)
+        except NotImplementedError:
+            # allow backwards compatibility or persisted_values way
+            for persisted_var in self.persisted_values():
+                if persisted_var in data:
+                    self.logger.debug("Loaded value {} for attribute {}".format(
+                        data[persisted_var], persisted_var))
+                    # Set the loaded value to the attribute on this class
+                    setattr(self, persisted_var, data[persisted_var])
+        except:
+            # log exception while loading and let it continue
+            self.logger.exception(
+                "Failed to deserialize block with data: {}".format(data))
 
     def _save(self):
         """ Save the values to persistence
         """
         self.logger.debug("Saving to persistence")
-        # generate item to be persisted by gathering all variables
-        # to be persisted into a dictionary
-        item = {persisted_var: getattr(self, persisted_var)
-                for persisted_var in self.persisted_values()}
+        try:
+            data = self.serialize()
```

Comment:
We should add a check and warning here after serializing (same for deserializing) to check to see if they also defined `persisted_values`. If they provided implementations for serialize and persisted values we should log a warning indicating that the persisted values method will be ignored.

Reply:
Added relevant checks when instantiating a block